RestfulDropApi.cs

```diff
@@ -0,0 +1,124 @@
+    //https://opsstaging.www.visualstudio.com/en-us/docs/integrate/api/blobstore/blobs?branch=teams%2Fartifact%2Fversion2
+
+    public sealed class RestfulDropApi : IDropApi
+    {
+        public static readonly string APIVersionParam = "api-version";
+        private readonly string _pat;
+        private readonly ILogger _logger;
+
+        public RestfulDropApi(string pat, ILoggerFactory loggerFactory)
+        {
+            if (string.IsNullOrWhiteSpace(pat))
+            {
+                throw new ArgumentException("Non-empty PAT required.", nameof(pat));
+            }
+
+            _pat = pat;
+
+            _logger = loggerFactory.CreateLogger<RestfulDropApi>();
+        }
+        
+        
+        /// <summary>
+        /// Gets the manifest details
+        /// </summary>
+        /// <param name="manifestUri">The manifest uri.</param>
+        /// <param name="blobAPIVersion">The API version to use for the blob API.</param>
+        /// <param name="relativeRoot">The root path relative to the drop to retrieve.</param>
+        /// <returns>The manifest details.</returns>         
+        public async Task<IEnumerable<VstsFile>> GetVstsManifest(Uri manifestUri, string blobAPIVersion, 
+                                                                 string relativeRoot)
+        {
+            // dotnet core doesn't currently handle vsts redirects well. Poking both teams about it
+            // for now disable redirects.
+            var clientHandler = new HttpClientHandler() { AllowAutoRedirect = false };
+            using (var client = new HttpClient(clientHandler))
+            {
+                var base64EncodedString = Convert.ToBase64String(Encoding.UTF8.GetBytes("vstsdockerbuild:" + _pat));
+                //Todo Polly for retries
+                _logger.LogDebug($"asking for drop manifest at {manifestUri}");
+                
+                var manifestRequest = new HttpRequestMessage(HttpMethod.Get, manifestUri);
+                manifestRequest.Headers.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Basic", base64EncodedString);
+                
+                var manifestResponse = await client.SendAsync(manifestRequest);
+                if (manifestResponse.StatusCode == HttpStatusCode.NotFound)
+                {
+                    throw new DropException($"VSTS drop not found: {manifestUri}");
+                }                
+
+                if (manifestResponse.StatusCode == HttpStatusCode.RedirectMethod ||
+                    manifestResponse.StatusCode == HttpStatusCode.Redirect)
+                {
+                     manifestResponse = await client.GetAsync(manifestResponse.Headers.Location);
+                }
+                manifestResponse.EnsureSuccessStatusCode();
+                string manifestContent = await manifestResponse.Content.ReadAsStringAsync();
```

Comment:
maybe be more efficient to parse the stream. These manifests can get pretty big. Also below with sas urls.

Reply:
TODO in future